deep_quoridor/src/quoridor.py

```diff
@@ -192,35 +192,31 @@ def can_place_wall(self, player: Player, position: tuple[int, int], orientation:
 
         return (self._grid[self._get_wall_slice(position, orientation)] == Board.FREE).all()
 
+    def _are_adjacent(self, position1: tuple[int, int], position2: tuple[int, int]) -> bool:
+        dr = position1[0] - position2[0]
+        dc = position1[1] - position2[1]
+        return ((dr == 1 or dr == -1) and dc == 0) or ((dc == 1 or dc == -1) and dr == 0)
+
     def is_wall_between(self, position1: tuple[int, int], position2: tuple[int, int]) -> bool:
         """
         Check if there is a wall between two positions.
         """
-        assert np.sum(np.abs(np.subtract(position1, position2))) == 1, "Positions must be adjacent"
-
-        position1_on_board = self.is_position_on_board(position1)
-        position2_on_board = self.is_position_on_board(position2)
-        assert position1_on_board or position2_on_board, "At least one position must be on the board"
+        assert self._are_adjacent(position1, position2), "Positions must be adjacent"
 
-        if position1_on_board and position2_on_board:
-            wall_position = (position1[0] + position2[0] + 2, position1[1] + position2[1] + 2)
```

Comment:
Checking that the wall position is in bounds instead of checking both positions is a nice improvement here. I did a little testing and found that there's a slight but noticeable improvement from removing the overhead of a function call.